### ISSUE TITLE
In a list with versioning enabled, attachment changes are not versioned

### ISSUE THREAD

**Issue by zshane15** [MicrosoftDocs/microsoft-365-community#404]
The trivia item "In a list with versioning enabled, attachment changes are not versioned" should be revised.

This documentation appears to be incorrect or misleading. When versioning is turned on for a list, newly added attachments always produce another version when saving. This can be seen when adding a new item or updating an existing item. SharePoint will create a version for the initial submission of the item, then a version when file is attached on the backend.

I have some Flows that are referencing version history to determine if the item should move forward in the process. However, the fact that version history is affected when uploading an attachment, is breaking my logic. I wish there was a way to determine a version which was created for the purposes of adding an attachment.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6080b47a-816c-8bc8-94c0-506d040ae9bd
* Version Independent ID: c06f1880-827d-1d0b-c0fc-601432749411
* Content: [Versioning in SharePoint](https://docs.microsoft.com/en-us/microsoft-365/community/versioning-basics-best-practices)
* Content Source: [Community/versioning-basics-best-practices.md](https://github.com/MicrosoftDocs/microsoft-365-community/blob/live/Community/versioning-basics-best-practices.md)
* Service: **sharepoint-online**
* GitHub Login: @PatD
* Microsoft Alias: **efrene**

**Comment by sympmarc** [comment]
@zshane15 - Have you looked at [Run a flow when a SharePoint column is modified](https://flow.microsoft.com/en-us/blog/run-a-flow-when-a-sharepoint-column-is-modified)? I haven't used that with the attachments column, but maybe it will solve your issue? Let us know if it does.

In the meantime, I'll chat with @PatD about rewording the article to make that line clearer.

**Comment by zshane15** [comment]
@sympmarc yes I tried getting that action to work but the versions created by attachments are really messing things up. There is no reliable way to programmatically return the last version which was created by the user changing data in relation to the current version.

For example, when an item is modified, I want to determine if the Status column changed. Without the any attachment changes I could simply just compare the triggerBody()?['Status'] to the previous version's Status. However, if the user uploaded 5 files and deleted 5 files, there are now 11 version created (1 for the data, 5 for removal, and 5 for upload).

In the version history data, there's nothing that says that a version was related to a file upload or removal. So when I try to use the Action for Get Changes, I have no way to know which version to compare it to.

The only way I can accomplish what I'm trying to do is just store a the previous version number in a separate column. But, at that point I might as well just store the Previous Status in a separate column (which is what I had to do). Not as elegant of a solution, but it works.

**Comment by sympmarc** [comment]
Maybe save the docs in a separate library rather than as attachments?
